ada_feeding/scripts/ada_planning_scene.py

```diff
@@ -653,15 +685,47 @@ def update_table_detection(self) -> None:
             )
             return
 
-        # Modify z position of table for planning scene object
-        # TODO: Need to figure out this value
-        detected_table_pose.pose.position.z -= 0.78
+        # Translate detected position of table into table's origin
+        detected_table_pose.pose.position.x += self.table_detection_offsets[0]
+        detected_table_pose.pose.position.y += self.table_detection_offsets[1]
+        detected_table_pose.pose.position.z += self.table_detection_offsets[2]
+
+        # Convert the default and latest table quaternions to pyquaternion objects
+        default_table_pyquat = Q(
+            w=self.objects[self.table_object_id].quat_xyzw[3],
+            x=self.objects[self.table_object_id].quat_xyzw[0],
+            y=self.objects[self.table_object_id].quat_xyzw[1],
+            z=self.objects[self.table_object_id].quat_xyzw[2],
+        )
+        latest_table_pyquat = Q(
+            w=detected_table_pose.pose.orientation.w,
+            x=detected_table_pose.pose.orientation.x,
+            y=detected_table_pose.pose.orientation.y,
+            z=detected_table_pose.pose.orientation.z,
+        )
+
+        # Calculate the absolute distance between the latest and default table quaternions
+        quat_dist = Q.absolute_distance(default_table_pyquat, latest_table_pyquat)
```

Comment:
I left a comment in the param file about how it would be nice if the threshold could have intuitive units. I looked into the documentation of this library, and this function doesn't specify what units it returns. 

Instead, I'd recommend you use [the formula from here](https://math.stackexchange.com/a/90098) to get theta (in radians). That is essentially the smallest rotation you need to get from one quaternion to the other. Then, the threshold also has intuitive units (specify in the param file that it is the threshold for the smallest rotation between the current quaternion and the default quaternion, in radians.)